src/tools/forwarder/forwarder.cpp

```diff
@@ -0,0 +1,213 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    This tool creates a terminating QUIC proxy to forward all incoming traffic
+    to a specified target.
+
+--*/
+
+#include "msquichelper.h"
+#include "msquic.hpp"
```

Comment:
If this serves as a sample code, do we really want to use cpp where the wrapper isn't officially supported?

Reply:
It's not necessarily a sample, but might actually grow to a real tool. Also, I have plans of eventually officially supporting the C++ wrapper.